### PR TITLE
DOC: Adding details on naming conventions and OO, procedural interfaces.

### LINKED ISSUE #1327: Document procedural and enum conventions is common convents.
In the [Common Conventions](https://simpleitk.readthedocs.io/en/master/conventions.html#) document the naming convention for the mapping from class name to the procedural name.

Also Document the naming conventions for Enums and how to accessed them in different languages.

### PR DESCRIPTION
**Pull request by zivy**
closes #1327

### CODE REVIEW

File: docs/source/conventions.rst [hunk 1]
```diff
@@ -3,6 +3,16 @@ Common Conventions
 
 This document contains common conventions that SimpleITK filters, and objects follow. It is intended to describe the interfaces that users should use and developers should implement. If a method or class does not specify different behavior or default values then those described here, it should be assumed that it follows the following conventions.
 
+Naming
+......
+
+The object oriented interface of process objects are named based on the specific operation they perform on their input image(s)
+with a suffix of **ImageFilter**, the ITK convention. For example, `AbsImageFilter <https://simpleitk.org/doxygen/latest/html/classitk_1_1simple_1_1AbsImageFilter.html>`_, `BinaryMinMaxCurvatureFlowImageFilter <https://simpleitk.org/doxygen/latest/html/classitk_1_1simple_1_1BinaryMinMaxCurvatureFlowImageFilter.html>`_ etc. For almost all process objects there is also a procedural interface. One or more functions which
```
Comment (dave3d) [inline docs/source/conventions.rst:10]: ^interface. One^interface, one^

The "One ..." sentence isn't a complete sentence.

File: docs/source/conventions.rst [hunk 2]
```diff
@@ -3,6 +3,16 @@ Common Conventions
...
+allow you to set parameter values and invoke the operation in one call. This results in concise code and is useful if you only need to
```
Comment (dave3d) [inline docs/source/conventions.rst:11]: ^you^a developer^

I'm not a fan of "you" unless it's really called for.

File: docs/source/conventions.rst (hunk 1 above)
Comment (zivy) [reply docs/source/conventions.rst]: Fixed.

File: docs/source/conventions.rst (hunk 2 above)
Comment (zivy) [reply docs/source/conventions.rst]: Removed, though was initially considering changing it to ewe, even sheep can use the procedural interface 😉 .

**Review by blowekamp — APPROVED** [review]

**Pull request merged**